test/integration/annotation/annotated_resource_setup.go

```diff
@@ -0,0 +1,318 @@
+package annotation
+
+import (
+	"context"
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/client"
+	"github.com/skupperproject/skupper/pkg/kube"
+	"github.com/skupperproject/skupper/pkg/utils"
+	"github.com/skupperproject/skupper/test/utils/k8s"
+	"gotest.tools/assert"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	nsCluster1 = "annotation-1"
+	nsCluster2 = "annotation-2"
+	timeout    = 120 * time.Second
+	interval   = 20 * time.Second
+)
+
+var (
+	clusters           int = 1
+	ctx                    = context.Background()
+	kubeConfigCluster1 string
+	kubeConfigCluster2 string
+	cluster1           *client.VanClient
+	cluster2           *client.VanClient
+	// teardownClusters prevents deleting an existing namespace
+	teardownClusters []*client.VanClient
+)
+
+// test allows defining the matrix to run the test table
+type test struct {
+	name                  string
+	doc                   string
+	modification          func(*testing.T)
+	expectedSites         int
+	expectedServicesProto map[string]string
+}
+
+// Setup simply creates the namespaces using a single
+// cluster by default, or using two clusters if at
+// least two kubeconfig files have been provided. It
+// also initializes the cluster1 and cluster2 variables.
+func Setup(t *testing.T) {
+	var err error
+
+	// Need at least one public (or a default) cluster
+	pubClusters := k8s.KubeConfigFilesCount(t, false, true)
+	if pubClusters == 0 {
+		t.Skip("no public clusters available")
+	}
+	if k8s.KubeConfigFilesCount(t, true, true) > 1 {
+		clusters = 2
+	}
+	t.Logf("Setup - using %d clusters", clusters)
+
+	// Initializing clients
+	t.Log("Initializing clients")
+
+	// cluster1 will be used as public (connection token will be generated from it)
+	kubeConfigCluster1 = k8s.KubeConfigFiles(t, false, true)[0]
+	// cluster2 will connect to cluster1 (can be either an edge, a public or the default)
+	kubeConfigCluster2 = k8s.KubeConfigFiles(t, true, true)[0]
+	cluster1, err = client.NewClient(nsCluster1, "", kubeConfigCluster1)
+	assert.Assert(t, err)
+	cluster2, err = client.NewClient(nsCluster2, "", kubeConfigCluster2)
+	assert.Assert(t, err)
```

Comment:
Correct, same like the other tests using --use-cluster are doing. But as I said above, the only reason for not reusing
was because of the possibility of using different clusters.

I totally agree we need to standardize and prioritize #209.